src/test/scala/chiselTests/experimental/ProgrammaticPortsSpec.scala

```diff
@@ -39,6 +40,23 @@ class ProgrammaticPortsSpec extends ChiselFlatSpec {
     assert(module.getNameFailures() == Nil)
   }
 
+  "ExtModule" should "support suggestName on ports" in {
+    lazy val module = new MultiIOModule {
+      val ext_module = Module(new ExtModule with NamedModuleTesterBase {
+        val clock = IO(Input(Clock()))
+        val reset = IO(Input(AsyncReset()))
+        val wdata  = IO(Input(UInt(8.W)))
+        val rdata  = IO(Output(UInt(8.W)))
+        expectName(clock.suggestName("clk"), "clk")
+        expectName(reset.suggestName("rst"), "rst")
+        expectName(wdata.suggestName("foo"), "foo")
+        expectName(rdata.suggestName("bar"), "bar")
+      })
+    }
+    elaborate { module }
+    assert(module.ext_module.getNameFailures() == Nil)
+  }
+
```

Comment:
This is kind of in conflict with the `LegacyModule` test below:
```scala

  "LegacyModule" should "ignore suggestName on ports" in {
    doTest(new Module with NamedModuleTester {
      val io = IO(new Bundle {
        val foo = Output(UInt(8.W))
      })
      expectName(io.suggestName("cheese"), "io")
      expectName(clock.suggestName("tart"), "clock")
      expectName(reset.suggestName("teser"), "reset")
    })
  }
```
I'm not really sure what we should do here, but ports have been something we've been a bit loathe to allow arbitrary naming of. I'm not saying we shouldn't do it though, just that we have avoided it in the past. The only `suggestName` on ports that's currently supported I think is when they aren't a public val in the class (see the "PrivatePort" test above).

Reply:
Sorry I reply so late, I think that **BlackBox** act as **LegacyModule** which restricted by `val io`, **ExtModule** act as **RawModule** which would be flexible. So **ExtModule** can rename IO doesn't conflict with **LegacyModule**.